sph/include/sph/positions.hpp

```diff
@@ -38,6 +38,41 @@
 
 namespace sph
 {
+template<class T>
+T fbcCheck(T x, T y, T z, T vx, T vy, T vz, T h, const cstone::Box<T> box)
+{
+    if (box.fbcX())
+    {
+        T distXmax = std::abs(box.xmax() - x);
+        T distXmin = std::abs(box.xmin() - x);
+
+        if (distXmax < 2.0 * h || distXmin < 2.0 * h)
+        {
+            if (vx == T(0.0) && vy == vx && vz == vx) { return T(0.0); }
+        }
+    }
+    if (box.fbcY())
+    {
+        T distYmax = std::abs(box.ymax() - y);
+        T distYmin = std::abs(box.ymin() - y);
+
+        if (distYmax < 2.0 * h || distYmin < 2.0 * h)
+        {
+            if (vx == T(0.0) && vy == vx && vz == vx) { return T(0.0); }
+        }
+    }
+    if (box.fbcZ())
+    {
+        T distZmax = std::abs(box.zmax() - z);
+        T distZmin = std::abs(box.zmin() - z);
+
+        if (distZmax < 2.0 * h || distZmin < 2.0 * h)
+        {
+            if (vx == T(0.0) && vy == vx && vz == vy) { return T(0.0); }
+        }
+    }
```

Comment:
code bloat: the three if statements can all be expressed with just one function and different arguments